### PR TITLE
Update https_everywhere_checker/rules.py, Fix #14290

### LINKED ISSUE #14290: Test if test url is covered by any target
Test urls with hosts not related to any target are currently allowed in rulesets.
The following scenario should should fail in ruleset tests:
```
<ruleset name="example">
	<target host="target.example" />

	<test url="http://no-target.example/" />

	<rule from="^http:" to="https:" />
</ruleset>
```

Type: other

### PR DESCRIPTION
**Pull request by cschanaj**
Make Travis to assert that each test url is covered by at least one target

- [x] Amazon-Adsystem.com.xml (#14357)
- [x] Amazon.xml (#14327)
- [ ] AmazonWebServices.xml (#14457)
- [x] Cloudinary.xml (#14430)
- [x] CrossRef.org.xml (#14294)
- [x] General-anzeiger-bonn.de.xml (#14300)
- [x] Gtimg.cn.xml (#14358)
- [ ] Hatena.xml
- [x] LiveJournal.com.xml (#14301)
- [x] ManageEngine.jp.xml (#14402)
- [x] Massengeschmack.tv.xml (#14302)
- [x] Migros-Bank.xml (#14340)
- [x] OpenNIC.org.xml (#14309)
- [x] Qpic.cn.xml (#14337)
- [x] Sikur.xml (#14338)
- [x] Sohu.com.xml (#14339)
- [x] Verizon.xml (#14429)
- [ ] Wemfbox.ch.xml (#14330)
- [x] XVideos.com.xml (#14341)
- [x] Zwame.pt.xml (#14342)
- [x] amazon.ca.xml (#14327)
- [x] amazon.co.jp.xml (#14327)
- [x] amazon.co.uk.xml (#14327)
- [x] amazon.com.au.xml (#14327)
- [x] amazon.com.br.xml (#14327)
- [x] amazon.com.mx.xml (~#14328~, #14327)
- [x] amazon.de.xml (#14327)
- [x] amazon.es.xml (#14327)
- [x] amazon.fr.xml (#14327)
- [x] amazon.it.xml (#14327)
- [x] amazon.nl.xml (#14327)
- [x] centrum.cz.xml (#14299)
- [ ] freetalklive.com.xml
- [x] lofter.xml (#14297)
- [x] naukimg.com.xml (#14403)
- [x] polyvoreimg.com.xml (#14343)
- [x] pornBB.xml (#14359)
- [x] thehubpeople.com.xml (#14360)
- [x] uptodown.net.xml (#14296)
- [x] webcaster.pro.xml (#14428)

### CODE REVIEW

**Comment by Bisaloo** [comment]
What should we do about the current rulesets failing this new test?

**Comment by J0WI** [comment]
At least all Amazon related rulesets are affected because of HSTS prune.

**Comment by cschanaj** [comment]
IMHO, we should update the rulesets to include to missing targets (if they are not removed because of hsts-prune). Luckily, they are not too difficult to fix...

File: test/rules/src/https_everywhere_checker/rules.py [hunk 1]
```diff
@@ -179,6 +179,37 @@ def _determineTestApplication(self):
 					self.test_application_problems.append("%s: No rule or exclusion applies to test URL %s" % (
 						self.filename, test.url))
 				self.determine_test_application_run = True
+
+			for test in self.tests:
+				urlParts = urlparse(test.url)
+				hostname = urlParts.hostname
+				ok = False
```
Comment (Hainish) [inline test/rules/src/https_everywhere_checker/rules.py]: Please be more descriptive in this variable name.  Also, for the first case, we can shorten it up:

```python
ok = hostname in self.targets
```

File: test/rules/src/https_everywhere_checker/rules.py [hunk 2]
```diff
@@ -179,6 +179,37 @@ def _determineTestApplication(self):
...
+							ok = True
+							break
+
+						parts[i] = tmp
+
+				if not ok:
+					self.test_application_problems.append("%s: No target applies to test URL %s" % (
+						self.filename, test.url))
+
+				self.determine_test_application_run = True
```
Comment (Hainish) [inline test/rules/src/https_everywhere_checker/rules.py]: Seems like this should be moved outside the loop, and same line in the above loop removed

**Comment by Hainish** [comment]
Tests for this should be fixed with https://github.com/EFForg/https-everywhere/pull/14521.

**Pull request merged**